lib/modules/datasource/generic.ts

```diff
@@ -0,0 +1,102 @@
+import is from '@sindresorhus/is';
+import jsonata from 'jsonata';
+import { z } from 'zod';
+import { logger } from '../../logger';
+import { readLocalFile } from '../../util/fs';
+import { parseUrl } from '../../util/url';
+import { Datasource } from './datasource';
+import type { ReleaseResult } from './types';
+
+export const ReleaseResultZod = z.object({
+  releases: z.array(
+    z.object({
+      version: z.string(),
+      isDeprecated: z.boolean().optional(),
+      releaseTimestamp: z.string().optional(),
+      sourceUrl: z.string().optional(),
+      sourceDirectory: z.string().optional(),
+      changelogUrl: z.string().optional(),
+    })
+  ),
+  sourceUrl: z.string().optional(),
+  sourceDirectory: z.string().optional(),
+  changelogUrl: z.string().optional(),
+  homepage: z.string().optional(),
+});
```

Comment:
Extract to `datasource/schema.ts` and rename to `ReleaseResult`. Let's not replace all instances of the `ReleaseResult` type with `z.infer<typeof ReleaseResult>` (yet) cause that would be outside the scope of this PR.

Reply:
This is currently only a subset of the full ReleaseResult, not sure if we want to expose the object.